nhs/data/filter.py

```diff
@@ -1,26 +1,147 @@
-from typing import List
+import glob
 
 import polars as pl
 
 
+def load_gnaf_files_by_states(
+    gnaf_path: str, states: list[str] = []
+) -> tuple[pl.LazyFrame, pl.LazyFrame]:
+    """
+    Load and filter ADDRESS_DEFAULT_GEOCODE and ADDRESS_DETAIL files for the specified states
+    from the given GNAF directory, and return them as LazyFrames.
+
+    Parameters:
+    - gnaf_path: str, the directory path where the GNAF psv files are stored.
```

Comment:
add documentation of what the file name should be inside this directory, since you are using `f"{gnaf_path}/{state_name}_ADDRESS_DEFAULT_GEOCODE_psv.psv"` later on

Reply:
and say what column each file in this directory should have, also we are not reading PSV files - you can detect the extension and use the appropriate reader function from `handling.py`

Reply:
missed this, also add name pattern of address detail files to the documentation since you have `{gnaf_path}/{state_name}_ADDRESS_DETAIL_psv.psv"`